apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/AbstractTraceHeaderPropagatingHandler.cs

```diff
@@ -25,41 +24,20 @@ namespace Google.Cloud.Diagnostics.Common
     /// Traces outgoing HTTP requests and propagates the trace header.
     /// </summary>
     ///
-    /// <example>
-    /// <code>
-    /// public void DoSomething(IManagedTracer tracer)
-    /// {
-    ///     var traceHeaderHandler = TraceHeaderPropagatingHandler.Create(tracer);
-    ///     using (var httpClient = new HttpClient(traceHeaderHandler))
-    ///     {
-    ///         ...
-    ///     }
-    /// }
-    /// </code>
-    /// </example>
     /// <remarks>
     /// Ensures the trace header is propagated in the headers for outgoing HTTP requests and 
     /// traces the total time of the outgoing HTTP request.  This is only done if tracing is initialized
     /// and tracing is enabled for the request current request.
     /// </remarks>
-    public class TraceHeaderPropagatingHandler : DelegatingHandler
+    public abstract class AbstractTraceHeaderPropagatingHandler : DelegatingHandler
```

Comment:
TraceHeaderPropagatingHandlerBase would be more idomatic here. But it's not entirely clear why it needs to be an abstract class at all.

Suppose we made it just TraceHeaderProagatingHandler and make the constructor or static factory method accept a `Func<IManagedTrace>` as a "current trace provider" would that work? I think it would do everything we need in a slightly simpler way.